Pfprototype: merged as requested and have run tests.

alicerobson opened the pull request.

Commit:
Added block builder

Commit:
tidying up of test case

Commit:
missing distance file

Commit:
fixes to allow different distance to be used

Commit:
just a test

Commit:
more tidying

cbernet committed:
fixes to be able to run on my mac

cbernet committed:
Merge branch 'alicerobson-pfprototype'

cbernet committed:
added comments and refactored code a bit with Alice

Commit:
implementing colins requests

Commit:
now has better block description

Commit:
updates to test case

Commit:
commenting and making structure nicer

Commit:
new test file for DAG

Commit:
and more sorting on block builder

Commit:
Added block splitter and tidied the blockbuilder accordingly

cbernet committed:
Merge pull request #8 from cbernet/master

bug fix: hadron shower prob in ecal now correct

Commit:
reconstructor added and is just about working

Commit:
reconstructor testing and fixing

Commit:
now compares two groups of reconstructed particles

Commit:
resolving merge conflicts

Commit:
working on comments and particle history

Commit:
now runs with pythia particles- new reconstruction matches original papas reconstruction

Commit:
working on comments

Commit:
more tidying and comments

Commit:
improving writing of history

Commit:
make reconstructed particles into a dict and created some history output

Commit:
make a merging tool separate from the reconstructor

Commit:
a little tidying

Commit:
and more tidying

Commit:
renamed files and fixed bug with cluster.is_inside

Commit:
Changes to make two rec methods match, including sorting and additional debug outputs

Commit:
even more ordering of tracks needed

Commit:
commenting and tidying

Commit:
tidying outputs

Commit:
merging with Colins master

Commit:
fixing a few minor things following merging

cbernet commented on test/pythis_tutorial.py:
```diff
@@ -0,0 +1,295 @@
+'''Example configuration file for an ee->ZH->mumubb analysis in heppy, with the FCC-ee
```
Now starting to comment! Thanks for the new PR.
I'll be a bit terse, thanks for your understanding :-)

typo in the name of this file pythis -> pythia

alicerobson replied:
done

cbernet commented on papas/aliceproto/PFfloodfill.py:
```diff
@@ -0,0 +1,82 @@
+
```
Why is the name of this module PFfloodfill.py? 
shouldn't we change that to floodfill.py ? 
Please note that it's difficult for me to understand what are the differences w/r to the module I had provided, because you are still using the aliceproto directory. I might have more comments when you do a further PR to replace my code.

alicerobson replied:
Sorry, this file is old and is not a part of my current python project. I hadn't spotted that it was still around and on git. It has now been removed. Also removed another "old" file in the same category.

cbernet commented on test/pythis_tutorial.py:
```diff
@@ -0,0 +1,295 @@
+        process(iev)
+    else:
+        loop.loop()
+        loop.write()
```
please add a newline

alicerobson replied:
done

cbernet commented:
I'm done commenting. Everything looks very good!
As soon as you are done taking my comments into account (or not if they are not relevant), 
I will merge, and we can start working towards removing aliceproto.

Thanks, let me know!

Commit:
addressing Colins comments

alicerobson commented:
Thanks for the comments Colin
I have addressed your comments and pushed the updated pfprototype branch up to my remote.

I have not moved any files out of  aliceproto (yet) but can start thinking about it
Alice

> On 05 Apr 2016, at 16:28, Colin Bernet notifications@github.com wrote:
> 
> I'm done commenting. Everything looks very good!
> As soon as you are done taking my comments into account (or not if they are not relevant), 
> I will merge, and we can start working towards removing aliceproto.
> 
> Thanks, let me know!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/cbernet/heppy/pull/10#issuecomment-205833275
